### PR TITLE
DOCSP-38100: User authentication

### PR DESCRIPTION
**Pull request by norareidy**
JIRA - https://jira.mongodb.org/browse/DOCSP-38100
Staging - https://preview-mongodbnorareidy.gatsbyjs.io/laravel/DOCSP-38100/user-authentication/
### Checklist

- [ ] Add tests and ensure they pass
- [ ] Add an entry to the CHANGELOG.md file
- [ ] Update documentation for new features

### CODE REVIEW

File: docs/includes/auth/AuthUser.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Models;
+
+use Jenssegers\Mongodb\Auth\User as Authenticatable;
+
+class User extends Authenticatable
```
Comment (GromNaN) [inline docs/includes/auth/AuthUser.php:7]: The developers don't have to create this class in their projects, it is available in the package: https://github.com/mongodb/laravel-mongodb/blob/4.6/src/Auth/User.php

Comment (norareidy) [reply docs/includes/auth/AuthUser.php:7]: I mention on the page that this class already exists, but they need to replace `use Illuminate\Foundation\Auth\User as Authenticatable` with `use Jenssegers\Mongodb\Auth\User as Authenticatable` - is that correct?

File: docs/user-authentication.txt [hunk 2]
```diff
@@ -11,14 +11,115 @@ User authentication
...
+
+.. code-block:: php
+   :emphasize-lines: 4
+
+   return [
+     App\Providers\AppServiceProvider::class,
+     MongoDB\Laravel\MongoDBServiceProvider::class,
+     MongoDB\Laravel\Auth\PasswordResetServiceProvider::class
+   ];
+
```
Comment (norareidy) [inline docs/user-authentication.txt:127]: I'm unsure about these instructions @GromNaN, is this the only step required to enable password reminders?

File: docs/includes/auth/AuthUser.php (hunk 1 above)
Comment (GromNaN) [reply docs/includes/auth/AuthUser.php:7]: Oh, sorry, I got confused. Indeed, they do need to create a `User` class that extends the one in our package. 
However, it is `MongoDB\Laravel\Auth\User` that should be used. `Jensseger` corresponds to the old namespace before we took over the package.

File: docs/includes/auth/AuthController.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
+namespace App\Http\Controllers;
```
Comment (GromNaN) [inline docs/includes/auth/AuthController.php:3]:
```suggestion
<?php

namespace App\Http\Controllers;
```

File: docs/user-authentication.txt [hunk 4]
```diff
@@ -11,14 +11,115 @@ User authentication
...
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+odm-long+} to modify the ``User``
+model and configure user authentication.
```
Comment (GromNaN) [inline docs/user-authentication.txt]: Modifying the User class is a technical solution, not the goal of the developer which is "Storing users in MongoDB when using Laravel's native auth  functionality".

File: docs/user-authentication.txt [hunk 5]
```diff
@@ -11,14 +11,115 @@ User authentication
...
+
+User Model
+----------
+
+By default, Laravel includes an ``App\Models\User`` Eloquent model in your ``app/Models``
```
Comment (GromNaN) [inline docs/user-authentication.txt]:
```suggestion
By default, Laravel generates the ``App\Models\User`` Eloquent model in your ``app/Models``
```

File: docs/user-authentication.txt [hunk 6]
```diff
@@ -11,14 +11,115 @@ User authentication
...
+directory. You can extend the ``Jenssegers\Mongodb\Auth\User`` class to configure your
+``User`` model to use the Laravel Auth module, which provides multiple authentication services.
```
Comment (GromNaN) [inline docs/user-authentication.txt]:
```suggestion
directory. You must extend the ``MongoDB\Laravel\Auth\User`` class to configure your
``User`` model to use MongoDB for the Laravel Auth module.
```

File: docs/user-authentication.txt [hunk 7]
```diff
@@ -11,14 +11,115 @@ User authentication
...
+To extend this class, navigate to your ``app/Models/User.php`` file and replace the
+``use Illuminate\Foundation\Auth\User as Authenticatable`` use statement with the following code:
+
+.. code-block:: php
+
+   use Jenssegers\Mongodb\Auth\User as Authenticatable;
```
Comment (GromNaN) [inline docs/user-authentication.txt]:
```suggestion
   use MongoDB\Laravel\Auth\User as Authenticatable;
```

File: docs/user-authentication.txt (hunk 2 above)
Comment (GromNaN) [reply docs/user-authentication.txt:127]: Yes, I think that's all you need to do. And I would like to get ride of this custom service declaration by making Laravel queries compatibles with MongoDB.

File: docs/includes/auth/AuthUser.php (hunk 1 above)
Comment (norareidy) [reply docs/includes/auth/AuthUser.php:7]: Ah got it, fixed!

File: docs/includes/auth/AuthController.php (hunk 3 above)
Comment (GromNaN) [reply docs/includes/auth/AuthController.php:3]: The `<?php` mark is required to be treated as PHP code.

**Review by GromNaN — APPROVED** [review]
LGTM

File: docs/user-authentication.txt [hunk 8]
```diff
@@ -11,14 +11,123 @@ User authentication
...
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to authenticate users stored in MongoDB
+by using Laravel's native authentication functionality.
```
Comment (mongoKart) [inline docs/user-authentication.txt]: S: applies throughout

```suggestion
In this guide, you can learn how to authenticate MongoDB users
by using Laravel's native authentication functionality.
```

File: docs/user-authentication.txt [hunk 9]
```diff
@@ -11,14 +11,123 @@ User authentication
...
+   :depth: 1
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to authenticate users stored in MongoDB
+by using Laravel's native authentication functionality. 
+
+Laravel provides a native Auth module that includes authentication services,
```
Comment (mongoKart) [inline docs/user-authentication.txt]: S: not sure whether this is standard for PHP/Laravel, but I'd guess it helps readability

```suggestion
Laravel provides a native ``Auth`` module that includes authentication services,
```

File: docs/user-authentication.txt [hunk 10]
```diff
@@ -11,14 +11,123 @@ User authentication
...
+
+Modify the User Model
+---------------------
+
+By default, Laravel generates the ``User`` Eloquent model in your ``App/Models``
+directory. To enable authentication for users stored in MongoDB, your ``User`` model
+must extend the ``MongoDB\Laravel\Auth\User`` class. 
+
+To extend this class, navigate to your ``app/Models/User.php`` file and replace the
+``use Illuminate\Foundation\Auth\User as Authenticatable`` use statement with the following
```
Comment (mongoKart) [inline docs/user-authentication.txt]:
```suggestion
``use Illuminate\Foundation\Auth\User as Authenticatable`` statement with the following
```

File: docs/user-authentication.txt [hunk 11]
```diff
@@ -11,14 +11,123 @@ User authentication
...
+code:
+
+.. code-block:: php
+
+   use MongoDB\Laravel\Auth\User as Authenticatable;
+
+Ensure that your ``User`` class extends ``Authenticatable``, as shown in the following code:
```
Comment (mongoKart) [inline docs/user-authentication.txt]: S: Might help to link this step to the previous one

```suggestion
Next, ensure that your ``User`` class extends ``Authenticatable``, as shown in the following code:
```

File: docs/user-authentication.txt [hunk 12]
```diff
@@ -11,14 +11,123 @@ User authentication
...
+
+.. code-block:: php
+
+   class User extends Authenticatable
+   {
+   ...
+   }
+
+After configuring your ``User`` model, create a corresponding controller. For instructions on
+creating a controller, see the :ref:`laravel-auth-controller` section on this page.
```
Comment (mongoKart) [inline docs/user-authentication.txt]: I: [on/about](https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#std-term-about:~:text=Use%20information%20about%2C%20not%20information%20on.)

```suggestion
After configuring your ``User`` model, create a corresponding controller. To learn how to
create a controller, see the :ref:`laravel-auth-controller` section on this page.
```

File: docs/user-authentication.txt [hunk 13]
```diff
@@ -11,14 +11,123 @@ User authentication
...
+
+Example
+~~~~~~~
+
+The following code demonstrates a ``User.php`` file that extends the ``MongoDB\Laravel\Auth\User``
```
Comment (mongoKart) [inline docs/user-authentication.txt]: S: shows? displays? "demonstrates" is more active to me

```suggestion
The following code shows a ``User.php`` file that extends the ``MongoDB\Laravel\Auth\User``
```

**Review by mongoKart — APPROVED** [review]
A few small changes and suggestions but LGTM otherwise

**Pull request merged**